src/View/Helper/PaginatorHelper.php

```diff
@@ -493,6 +507,21 @@ public function generateUrl(array $options = [], $model = null, $urlOptions = fa
             'fullBase' => false
         ];
 
+        return $this->Url->build($this->buildUrl($options, $model), $urlOptions);
+    }
+
+    /**
+     * Merges passed URL options with current pagination state to generate a pagination URL.
+     *
+     * @param array $options Pagination/URL options array
+     * @param string|null $model Which model to paginate on
+     * @return array An array of URL parameters
+     */
+    public function buildUrl(array $options = [], $model = null)
```

Comment:
it doesn't technically return a URL. it returns an array, so how about calling it `addUrlParams()`? Which is closer to what it does. (or something like that).